ipv8/test/messaging/interfaces/udp/test_endpoint.py

```diff
@@ -41,6 +77,10 @@ async def tearDown(self):
             self.endpoint2.close()
         await super(TestUDPEndpoint, self).tearDown()
 
+    def test_estimated_wan(self):
+        # The first address 10.0.0.2 is skipped as it has netmask 0.0.0.0
+        self.assertEqual(self.endpoint2_listener.my_estimated_wan[0], '192.168.15.1')
```

Comment:
The following needs to be addressed for this unit test:

 - The test is testing `EndpointListener` functionality and should therefore be in the`TestEndpointListener` test case, in the `ipv8/test/messaging/endpoint.py` file.
 - You changed `_get_interface_addresses()` and that method is not being called directly, making this test very fragile in case of future changes to any of the intermediate calls between `_get_interface_addresses()` and `my_estimated_wan`.
 - The arranging for this test is passive, providing this particular netmask case for all other units.
 - There is no inverted unit test to accompany this test (what if the netmask is not `0.0.0.0`?).
 - The unit test itself should have a single-line docstring explaining what the test does, preferably without reading the code of the test itself.

Reply:
I think I addressed all of these points:
* I put the test in a new file `ipv8/test/messaging/interfaces/test_endpoint.py` (note the `interfaces` subfolder, I think it is where the test should be placed, as `EndpointListener` is also inside the `interfaces` subfolder).
* Now the test tests the `_get_interface_addresses()` method directly.
* The test is local and does not change the behavior of other tests.
* The test checks both the absence of interface with netmask 0.0.0.0 and the presence of other interfaces.
* The docstring is added.

Reply:
Nice, thanks 👍